Separate plots using #### in make_room_for_ylabel_using_axesgrid.py

ardieorden opened the pull request:
## PR Summary

Separate plots using ######### lines for Sphinx-gallery to render plots in sections.
Part of #8922.

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

<!--
Thank you so much for your PR!  To help us review your contribution, please
consider the following points:

- A development guide is available at https://matplotlib.org/devdocs/devel/index.html.

- Help with git and github is available at
  https://matplotlib.org/devel/gitwash/development_workflow.html.

- Do not create the PR out of master, but out of a separate branch.

- The PR title should summarize the changes, for example "Raise ValueError on
  non-numeric input to set_xlim".  Avoid non-descriptive titles such as
  "Addresses issue #8576".

- The summary should provide at least 1-2 sentences describing the pull request
  in detail (Why is this change required?  What problem does it solve?) and
  link to any relevant issues.

- If you are contributing fixes to docstrings, please pay attention to
  http://matplotlib.org/devel/documenting_mpl.html#formatting.  In particular,
  note the difference between using single backquotes, double backquotes, and
  asterisks in the markup.

We understand that PRs can sometimes be overwhelming, especially as the
reviews start coming in.  Please let us know if the reviews are unclear or
the recommended next step seems overly demanding, if you would like help in
addressing a reviewer's comments, or if you have been waiting too long to hear
back on your PR.
-->

timhoffm commented on examples/axes_grid1/make_room_for_ylabel_using_axesgrid.py:
```diff
@@ -7,61 +7,65 @@
 
-if __name__ == "__main__":
+def ex1():
+    plt.figure(1)
```
```suggestion
    plt.figure()
```
We don't make any use of explicit figure numbers, so let's remove them to reduce clutter. Also in the other examples.

ardieorden commented:
Thanks for the feedback and apologies for the late response! I removed the functions wrote them as top-level code and I also removed the figure numbers for each code block. Let me know if you'd like to see other changes!

QuLogic commented on examples/axes_grid1/make_room_for_ylabel_using_axesgrid.py:
```diff
@@ -7,61 +7,56 @@
-    import matplotlib.pyplot as plt
+plt.figure()
+ax = plt.axes([0, 0, 1, 1])
+#ax = plt.subplot(111)
```
Might as well remove this.

timhoffm replied:
Just remove the commented line.

QuLogic commented on examples/axes_grid1/make_room_for_ylabel_using_axesgrid.py:
```diff
@@ -7,61 +7,56 @@
-    def ex3():
+###############################################################################
 
-        fig = plt.figure(3)
```
This `fig` is used later, but you've removed it.

ardieorden commented:
Thanks for the code review @QuLogic ! I committed the changes for importing matplotlib and assigning the `fig` variable and I squashed it to the previous commit for cleanliness.

Let me know if you'd like to see other changes!

ardieorden commented:
Thanks for the comment @timhoffm ! I just added in the `import matplotlib.pyplot as plt` line above all the other imports and I removed the commented out line. Let me know if you'd like to request for more changes!

timhoffm commented:
Thanks, and congratulations on your first contribution to Matplotlib! Hope to see you back some time.
